docs/preview/features/message-pumps/customization.md

```diff
@@ -76,6 +76,44 @@ public void ConfigureServices(IServiceCollection services)
 > Note that the order in which the message handlers are registered is important in the message processing.
 > In the example, when a message handler above this one is registered that could also handle the message (same message type) than that handler may be chosen instead of the one with the specific filter.
 
+## Filter messages based on message body
+
+When registereing a new message handler, one can opt-in to add a filter on the incoming message body which filters out messages that are not needed to be processed.
+This can be useful when you want to route messages based on the message content itself instead of the messaging context.
+
+Following example shows how a message handler should only process a certain message when it contains in its body the value 'Done'; meaning only done `Order`s will be processed.
+
+```csharp
+// Message to be sent:
+public enum Status { Processing, Done }
+
+public class Order
+{
+    public string Id { get; set; }
+    public string Status { get; set; }
+}
+
+// Message handler
+public class OrderMessageHandler : IMessageHandler<Order>
+{
+    public async Task ProcessMessageAsync(Order order, MessageContext context, ...)
+    {
+        // Do some processing...
+    }
+}
+
+// Message handler registration
+public class Startup
+{
+    ...
+    
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.WithMessageHandler<OrderMessageHandler, Order>(body => body.Contains("Done"));
```

Comment:
Maybe we should expand this sample so that it uses JToken.Parse and uses token["status"] == "Done"? (or even change Status type to an enum OrderStatus)

Reply:
Adding @SamVanhoutte as he initially requested this feature

Reply:
Looks good to me.  Agree having a sample to subscribe on json-payload would be great.

Reply:
I'm wondering if we should use the typed-predicate here, as in `(Order order) => order.Status == OrderStatus.Done`). Internally, it's not that big a change and it eliminates the 'risk' that consumers have the parse the raw body themselves?

Reply:
Fine for me!

Reply:
Should we then still support the raw string predicate? Or only the typed predicate?

Reply:
If it's introduced in this PR then we can remove it, otherwise we have to still support it